### ISSUE TITLE
Overflow when computing data bounds for small signed integer data types

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#2513]
### Short description

Data bounds are calculated using their underlying data type. For signed small integer data types where the data spans more than half the full-scale range, an exception occurs during computation of the data bounds. 

### Code to reproduce

```python
import pyqtgraph as pg
import numpy as np
pg.plot(np.array([-20000, 20000], dtype=np.int16))
pg.exec()
```

### Expected behavior

ViewBox should be auto-scaled around the data bounds.

### Real behavior

```
c:\work\repos\pyqtgraph\pyqtgraph\graphicsItems\ViewBox\ViewBox.py:1464: RuntimeWarning: overflow encountered in short_scalars
  bounds = QtCore.QRectF(xr[0], yr[0], xr[1]-xr[0], yr[1]-yr[0])
```

### Tested environment(s)

 * PyQtGraph version: master
 * Qt Python binding: PyQt6 6.3.1
 * Python version: Python 3.10.8
 * NumPy version: 1.23.4
 * Operating system: Windows 11
 * Installation method: pip install -e .

### Additional context
This can of course be easily worked around by the user by simply converting the data to float32 prior to plot.

**Comment by j9ac9k** [comment]
Not actually fixed by #2515

**Comment by pijyoi** [comment]
The included reproducing example in this issue is indeed fixed by #2515.
